sim/midas/src/main/scala/midas/stage/GoldenGateFileEmission.scala

```diff
@@ -0,0 +1,46 @@
+// See LICENSE for license details.
+
+package midas.stage
+
+import firrtl.annotations.{NoTargetAnnotation, Annotation}
+import firrtl.options.{CustomFileEmission}
+
+import chisel3.experimental.{annotate, ChiselAnnotation}
+
+trait GoldenGateFileEmission extends CustomFileEmission { this: Annotation =>
+  override def baseFileName(annotations: firrtl.AnnotationSeq) = {
+    annotations.collectFirst{ case OutputBaseFilenameAnnotation(name) => name }.get
+  }
+}
+
+/**
+  * A generic wrapper for output files that have no targets.
+  *
+  * @param body the body of the file
+  * @param fileSuffix The string to append to base output file name
+  *
+  */
+case class GoldenGateOutputFileAnnotation(body: String, fileSuffix: String)
+    extends NoTargetAnnotation with GoldenGateFileEmission {
+  def suffix = Some(fileSuffix)
+  def getBytes = body.getBytes
+}
+
+object GoldenGateOutputFileAnnotation {
+  /**
+    * Sugar to add a new output file from a chisel source (e.g., in a bridge, platform shim)
+    */
+  def annotateFromChisel(body: String, fileSuffix: String): Unit = {
+    annotate(new ChiselAnnotation { def toFirrtl = GoldenGateOutputFileAnnotation(body, fileSuffix) })
+  }
+}
```

Comment:
It looks like this: https://github.com/firesim/firesim/pull/802/files#diff-43e77e02a0cf4dfa093ddf410a9ac665b16f8be6c7e1eec75cdc790155728517R269

Frankly i detest the pattern of explicitly naming Chisel annotation classes that essentially just wrap a FIRRTL annotation. Instead, let's get the name of the method right, and possibly put it in an object with a better name (then it could just be an apply method).

Reply:
> Frankly i detest the pattern of explicitly naming Chisel annotation ...

fair.  And, this is probably the strongest reaction I've ever seen out of you.  Take it easy... ;)